fileredactor/filepathprocessor.go

```diff
@@ -0,0 +1,75 @@
+package fileredactor
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+)
+
+// FilePathProcessor is an interface for an entity which accepts file paths separated by the
+// newline (`\n`) character and returns a slice of absolute paths.
+type FilePathProcessor interface {
+	ProcessFilePaths(string) ([]string, error)
+}
+
+type filePathProcessor struct {
+	envRepository env.Repository
+	pathModifier  pathutil.PathModifier
+	pathChecker   pathutil.PathChecker
+}
+
+// NewFilePathProcessor returns a structure which implements the FilePathProcessor interface.
+// The implementation includes handling filepaths defined as environment variables, relative file paths,
+// and absolute file paths.
+// The implementation also includes making sure the filepath exists and is not a directory.
+func NewFilePathProcessor(repository env.Repository, modifier pathutil.PathModifier, checker pathutil.PathChecker) FilePathProcessor {
+	return filePathProcessor{
+		envRepository: repository,
+		pathModifier:  modifier,
+		pathChecker:   checker,
+	}
+}
+
+func (f filePathProcessor) ProcessFilePaths(filePaths string) ([]string, error) {
+	filePaths = strings.TrimSpace(filePaths)
+	if filePaths == "" {
+		return nil, nil
+	}
+
+	var processedFilePaths []string
+
+	list := strings.Split(filePaths, "\n")
+	for _, item := range list {
+		item = strings.TrimSpace(item)
+		if item == "" {
+			continue
+		}
+
+		path := item
+		if strings.HasPrefix(item, "$") {
+			path = f.envRepository.Get(item[1:])
+			if path == "" {
+				return nil, fmt.Errorf("invalid item (%s): environment variable isn't set", item)
+			}
+		}
+
+		path, err := f.pathModifier.AbsPath(path)
+		if err != nil {
+			return nil, err
+		}
+
+		isDir, err := f.pathChecker.IsDirExists(path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check if path (%s) is a directory: %w", path, err)
+		}
+		if isDir {
+			return nil, fmt.Errorf("path (%s) is a directory and cannot be redacted, please make sure to only provide filepaths as inputs", path)
```

Comment:
Is it a deliberate design decision that if one item is a dir, the entire function returns with an error?

Reply:
The recursive directory walking indeed sounds cool but I think it raises a couple of questions.

The redaction only works with text files so if someone uses a directory path then we need to collect all of the text files. We cannot do that based on file extension but I saw some tools which can tell you what kind of type a file is. 
Then there is a risk that users could use a single folder as the input (for example the deploy dir) and just dump everything in there. Even files which does not need redaction. This will increase the runtime of the step unnecessarily. 
Based on the above I think if we force the user to actually list all of the files is better because we do not need to add more logic to the step and also prevent the user making an error and accidentally increasing the step runtime.

And if the users really like this feature and come with feature request where they want to use folders as input value then we can always revisit this and introduce it as a non breaking change.

Reply:
I see, I forgot that the input could be non-text file too. I'm okay with the current implementation then!